orca-remote-stage/src/main/kotlin/com/netflix/spinnaker/orca/remote/pipeline/RemoteStage.kt

```diff
@@ -0,0 +1,37 @@
+package com.netflix.spinnaker.orca.remote.pipeline
+
+import com.netflix.spinnaker.kork.annotations.Beta
+import com.netflix.spinnaker.orca.api.pipeline.graph.StageDefinitionBuilder
+import com.netflix.spinnaker.orca.api.pipeline.graph.StageDefinitionBuilder.Aliases
+import com.netflix.spinnaker.orca.api.pipeline.graph.TaskNode
+import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution
+import com.netflix.spinnaker.orca.remote.model.RemoteStageExtensionConfig
+import com.netflix.spinnaker.orca.remote.service.RemoteStageExtensionService
+import com.netflix.spinnaker.orca.remote.tasks.MonitorRemoteStageTask
+import com.netflix.spinnaker.orca.remote.tasks.StartRemoteStageTask
+import org.springframework.stereotype.Component
+
+@Beta
+@Component
+@Aliases("remoteStage")
+class RemoteStage(
+  private val remoteStageExtensionService: RemoteStageExtensionService
+) : StageDefinitionBuilder {
+
+  override fun taskGraph(stage: StageExecution, builder: TaskNode.Builder) {
+    val remoteExtensionConfig = remoteStageExtensionService
+      .getByStageType(stage.type)
+      .getTypedConfig<RemoteStageExtensionConfig>()
+
+    // Set default parameters if necessary
+    remoteExtensionConfig.parameters.forEach { (k, v) ->
+      if (stage.context[k] == null) {
+        stage.context[k] = v
+      }
```

Comment:
Why do we only set parameters if the context doesn't have the value? Wouldn't we always want the config to be applied to the stage?

Reply:
No, I think we only want to use the defaults from the config if the stage is not run with the same parameters and values already defined in the context.  Does that make sense?